src/Nethermind/Nethermind.AccountAbstraction/Executor/UserOperationSimulator.cs

```diff
@@ -98,13 +98,13 @@ public class UserOperationSimulator : IUserOperationSimulator
                 }
             }
 
-            IReleaseSpec currentSpec = _specProvider.GetSpec(parent.Number + 1);
+            IReleaseSpec specfor1559 = _specProvider.GetSpecFor1559(parent.Number + 1);
```

Comment:
@smartprogrammer93 Why not _specProvider.IsEip1599Enabled(blockNumber)
Could we remove public method GetSpec(blockNumber)?

Reply:
I see, because of Eip1559TransitionBlock, and a few other properties. Just wondering how we can avoid mistakes like_specProvider.GetSpec(byNumberOnly) in future.

Reply:
Addressed in the latest changes.

Reply:
Don't we need a proper ReleaseSpec here with timestamp forks too?